src/Computation.jl

```diff
@@ -0,0 +1,251 @@
+  sampleMultinomial(word_ind, document, state)
+end
+
+
+function updateSufficientStatistics(word::Int64,
+                                    topic::Int64,
+                                    document::Int64,
+                                    scale::Float64,
+                                    state::State)
+  fr = Float64(!state.frozen)
+  @inbounds state.docSums[topic, document] += scale
+  @inbounds state.topicSums[topic] += scale * fr
+  @inbounds state.topics[topic, word] += scale * fr
+  return
+end
+
+@doc raw"""
+    getTermDist(state::State, model::Model)
+
+Compute ``\phi_{k,v} = \frac{\Psi_{k,v} + \beta_t}{\left( \sum^V_{v'=1} \Psi_{k,v'} + \beta_{v'}\right)}``
+
+Where ``\vec{ϕ_v}`` parameterizes the V-dimensional categorical distribution of a word.
+
+Updates the `termDist` attribute of `state`
+"""
+function getTermDist(state::State, model::Model)
+  Phi = Array{Float64,2}(undef,length(model.alphaPrior),length(model.betaPrior))
+  for topic in 1:length(model.alphaPrior)
+    Phi[topic,:] = (state.topics[topic,:] .+ model.betaPrior) ./ (state.topicSums[topic] + sum(model.betaPrior))
+  end
+  return Phi
+end
+
+@doc raw"""
+    getTopicDist(state::State, model::Model)
+
+Compute ``\theta_{k,m} = \frac{\Omega_{k,m} + \alpha_k}{\left( \sum^K_{k'=1} \Omega_{k,m'} + \alpha_{k'}\right)}``
+
+Where ``\vec{\theta_m}`` parameterizes the K-dimensional categorical distribution of a document.
+
+Updates the `topicDist` attribute of `state`
+"""
+function getTopicDist(state::State, model::Model)
+  Theta = Array{Float64,2}(undef,length(model.alphaPrior),length(model.corpus))
+  for doc in 1:length(model.corpus)
+    Theta[:,doc] = (state.docSums[:,doc] .+ model.alphaPrior) ./ (sum(state.docSums[:,doc]) + sum(model.alphaPrior))
+  end
+  return Theta
+end
+
+function sampleDocument(document::Int,
+                        model::Model,
+                        state::State)
+  words = model.corpus.docs[document].terms
+  Nw = length(words)
+  @inbounds weights = model.corpus.weights[document]
+  K = length(model.alphaPrior)
+  @inbounds assignments = state.assignments[document]
+  for ii in 1:Nw
+    word = words[ii]
+    oldTopic = assignments[ii]
+    updateSufficientStatistics(word, oldTopic, document, -weights[ii], state)
+    newTopic = sampleWord(word, ii, document, model, state)
+    @inbounds assignments[ii] = newTopic
+    updateSufficientStatistics(word, newTopic, document, weights[ii], state)
+  end
+  return
+end
+
+function sampleCorpus(model::Model, state::State)
+  for ii in 1:length(model.corpus)
+    sampleDocument(ii, model, state)
+  end
+  return
+end
+
+# The functions below are designed for public consumption
+function trainModel(model::Model,
+                    state::State,
+                    numIterations::Int64)
+  for ii in 1:numIterations
+    println(string("Iteration ", ii, "..."))
+    sampleCorpus(model, state)
+  end
+  return
+end
+
+function topTopicWords(model::Model,
```

Comment:
I would consider this moving to a separate file where someday all the functions which visualize/inspect the model could live.